code/common/src/test/scala/com/avast/cloud/datadog4s/helpers/RepeatedTest.scala

```diff
@@ -19,16 +19,38 @@ class RepeatedTest extends AnyFlatSpec with Matchers {
   private val noopErrHandler: Throwable => IO[Unit] = (_: Throwable) => IO.unit
 
   "repeated test" should "be called repeatedly" in {
-    val test = Ref.of[IO, Int](0).flatMap { ref =>
-      val forever =
-        Repeated.run[IO](Duration.ofMillis(5), Duration.ofMillis(50), noopErrHandler) {
-          IO.delay(logger.info("increasing ref")) *> ref.update(_ + 1)
-        }
-      forever.use(_ => IO.never).timeout(100 milli).attempt.flatMap(_ => ref.get)
+    val waitFor = 10
+
+    def buildProcess(counter: Ref[IO, Int], killSignal: Deferred[IO, Unit]): IO[Int] = {
+      def decreaseCounter: IO[Unit] =
+        counter.modify { currentCount =>
+          currentCount - 1 match {
+            case newCounter if newCounter <= 0 => (newCounter, killSignal.complete(()))
+            case newCounter                    => (newCounter, IO.pure(()))
+          }
+        }.flatMap(identity)
```

Comment:
`.flatten`

Reply:
io doesn't have flatten

Reply:
🤦 `flatten` is on `flatmap`, not `monad` ...